src/table.py

```diff
@@ -1932,11 +1959,22 @@ def are_neighbors(r1, r2):
         This check supports empty rect-likes and thus also lines.
         """
         if (
-            r2.x0 - snap_x <= r1.x0 <= r2.x1 + snap_x
-            or r2.x0 - snap_x <= r1.x1 <= r2.x1 + snap_x
-        ) and (
-            r2.y0 - snap_y <= r1.y0 <= r2.y1 + snap_y
-            or r2.y0 - snap_y <= r1.y1 <= r2.y1 + snap_y
+            (
+                r2.x0 - snap_x <= r1.x0 <= r2.x1 + snap_x
+                or r2.x0 - snap_x <= r1.x1 <= r2.x1 + snap_x
+            )
+            and (
+                r2.y0 - snap_y <= r1.y0 <= r2.y1 + snap_y
+                or r2.y0 - snap_y <= r1.y1 <= r2.y1 + snap_y
+            )
+            or (
+                r1.x0 - snap_x <= r2.x0 <= r1.x1 + snap_x
+                or r1.x0 - snap_x <= r2.x1 <= r1.x1 + snap_x
+            )
+            and (
+                r1.y0 - snap_y <= r2.y0 <= r1.y1 + snap_y
+                or r1.y0 - snap_y <= r2.y1 <= r1.y1 + snap_y
+            )
```

Comment:
I don't know what it highest priority, `and` or `or`. Could we make this explicit with brackets here? I.e. instead of `if (expr1) and (expr2) or (expr3) and (expr4):`, suggest either `if ((expr1) and (expr2)) or ((expr3) and (expr4)):`, or `if (expr1) and ((expr2) or (expr3)) and (expr4):`.

Reply:
You don't? Well I can of course add more brackets ...